python38 executors

akshaps opened the pull request:
Since Python 3.6 End-Of-Life (EOL) reached on December 23rd, 2021, Lambda will no longer apply security patches and other updates to the Python 3.6 runtime used by Lambda functions, and functions using Python 3.6 will no longer be eligible for technical support from July 18th, 2022 onwards. 

Hence, we're upgrading our existing Python 3.6 functions to Python 3.8 for now.

Commit:
python38 executors

bondnotanymore commented on src/jobs/sam-build.yml:
```diff
@@ -6,7 +6,7 @@ parameters:
     description: Version of python to use with pyenv
     type: string
-    default: "3.6.11"
+    default: "3.8.12"
```
I think we already discussed that we wont touch the existing jobs but rather use their parameters to call them with a parameter value of our choice

We shouldn't be updating sam-build job's default python version direclty here as it can lead to backward compatibility issues for other consumers of this orb.

Rather, we should call the sam-build job with the pyenv-version parameter specified of our choice from the service using the orb.

akshaps replied:
taken care in this commit [a5611debe4df37e68d70a6c2b72b51437122a55f](https://github.com/rackerlabs/rsspca/pull/33/commits/a5611debe4df37e68d70a6c2b72b51437122a55f)

Commit:
reverting the version to 3.6.11 from 3.8.12
As the version 3.8.12 can be explicitly set as a parameter wherever needed, so as to not to disturb the existing jobs

catogami commented:
BotComment: *Production* version of orb available for use - `rackerlabs/rsspca@0.14.0`
